Add Bicep Deployment for Hub and Spoke Hack

mbrat2005 opened the pull request:
# What the Hack Networking Bicep Deployment

This directory contains Bicep templates to deploy and configure resources as described in each WTH Networking challenge. These deployments represent one way to meet the challenge requirements--there are many others.

## Who should use these templates?

The WTH philosophy intends to have students learn by doing, and recognizes that one of the best ways to learn is to troubleshoot problems. As such, using these templates instead of building your own lab will detract from your learning experience, and primarily recommended for the scenarios below: 

- Students who will not be completing a challenge which is a prerequisite to a later challenge
- Students who are falling behind in the WTH due to issues unrelated to the core learning goals of this WTH
- Students looking for a reference implementation to compare against their own approach

jrzyshr commented:
Hi @mbrat2005 Thanks for the contribution of these templates!  I've added @erjosito to review.

As per email shared within Microsoft, you should move these templates from /Student/Resources to the /Coach/Solutions folder.  
-Peter

mbrat2005 commented:
@jrzyshr, I made the requested changes:
- Moved the bicep content to the Coaches/Solution directory
- Removed .gitattributes change (created new .gitattributes in the DOCKERFILE location)
- Added documentation to the Coaches readme with details on the Bicep solution and when it may be used
